packages/pn-validator/src/__test__/Validator.test.ts

```diff
@@ -79,8 +94,46 @@ describe('Test Validator', () => {
           propertyTwo: 'valueTwo',
         },
       ],
+      fooProperty: 'foo'
     };
     const results = dummyValidator.validate(dummyObject);
     expect(results).toBeNull();
   });
 });
+
+describe('Test Validator with strict mode enabled', () => {
+  const dummyObject: DummyClass = {
+    property: 'value',
+    subProperty: [
+      {
+        propertyOne: 'valueOne',
+        propertyTwo: 'valueTwo',
+      },
+    ],
+    fooProperty: 'foo',
+  };
+
+  it('should throw validation error for missing rules (strict: true)', () => {
+    const validatorWithOptions = new DummyFooValidator({ strict: true });
+    // Since a rule for 'fooProperty' and the properties of 'subProperty' (propertyOne and propertyTwo)
+    // are missing, an error is expected
+    const expectedError = {
+      subProperty: [
+        {
+          propertyOne: 'Rule is missing',
+          propertyTwo: 'Rule is missing',
+        },
+      ],
+      fooProperty: 'Rule is missing',
+    };
+    const results = validatorWithOptions.validate(dummyObject);
+    expect(results).toStrictEqual(results);
```

Comment:
You should check `expect(results).toStrictEqual(expectedError);` and not `expect(results).toStrictEqual(results);`.
This test will fail because the isMissingRule function doesn't do a deep check on the nested objects or arrays.
That's why we need two separate tests:
- this one that doesn't check nested properties
- another one in which nested validator has the strict option set to true